File: src/leetcode/P228/java/Solution.java
package leetcode.P228.java;

import java.util.ArrayList;
import java.util.List;

public class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> list = new ArrayList<>();

        int num = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] + 1 != nums[i + 1 < nums.length ? i + 1 : nums.length - 1]) {
                if (num == i) {
                    list.add("" + nums[i]);
                } else {
                    list.add("" + nums[num] + "->" + nums[i]);
                }
                num = i + 1;
            }
        }
        return list;
    }
}

File: src/leetcode/p228/kotlin/Solution.kt
package leetcode.p228.kotlin

class Solution {
    fun summaryRanges(nums: IntArray): List<String> {
        val list = ArrayList<String>()
        var num = 0

        for (i in nums.indices) {
            var checkIndex = if (i + 1 < nums.size) i + 1 else nums.size - 1
            if (nums[i] + 1 != nums[checkIndex]) {
                if (num == i) {
                    list.add("" + nums[i]);
                } else {
                    list.add("" + nums[num] + "->" + nums[i])
                }
                num = i + 1
            }
        }
        return list
    }
}